measure/src/test/scala/org/apache/griffin/measure/transformations/AccuracyTransformationsIntegrationTest.scala

```diff
@@ -0,0 +1,169 @@
+      sourceName = PERSON_TABLE,
+      targetName = PERSON_TABLE,
+      expectedResult = AccuracyResult(total = 2, miss = 0, matched = 2, matchedFraction = 1.0))
+  }
+
+  "accuracy" should "work with empty target" in {
+    checkAccuracy(
+      sourceName = PERSON_TABLE,
+      targetName = EMPTY_PERSON_TABLE,
+      expectedResult = AccuracyResult(total = 2, miss = 2, matched = 0, matchedFraction = 0.0)
+    )
+  }
+
+  "accuracy" should "work with empty source" in {
+    checkAccuracy(
+      sourceName = EMPTY_PERSON_TABLE,
+      targetName = PERSON_TABLE,
+      expectedResult = AccuracyResult(total = 0, miss = 0, matched = 0, matchedFraction = 1.0))
+  }
+
+  "accuracy" should "work with empty source and target" in {
+    checkAccuracy(
+      sourceName = EMPTY_PERSON_TABLE,
+      targetName = EMPTY_PERSON_TABLE,
+      expectedResult = AccuracyResult(total = 0, miss = 0, matched = 0, matchedFraction = 1.0))
+  }
+
+  private def checkAccuracy(sourceName: String, targetName: String, expectedResult: AccuracyResult) = {
+    val dqContext: DQContext = getDqContext(
+      dataSourcesParam = List(
+        DataSourceParam(
+          name = "source",
+          connectors = List(dataConnectorParam(tableName = sourceName))
+        ),
+        DataSourceParam(
+          name = "target",
+          connectors = List(dataConnectorParam(tableName = targetName))
+        )
+      ))
+
+    val accuracyRule = RuleParam(
+      dslType = "griffin-dsl",
+      dqType = "ACCURACY",
+      outDfName = "person_accuracy",
+      rule = "source.name = target.name"
+    )
+
+    val res = getRuleResults(dqContext, accuracyRule)
+      .as[AccuracyResult]
+      .collect()
+
+    res.length shouldBe 1
+
+    res(0) shouldEqual expectedResult
+  }
+
+  private def getRuleResults(dqContext: DQContext, rule: RuleParam): DataFrame = {
+    val dqJob = DQJobBuilder.buildDQJob(
+      dqContext,
+      evaluateRuleParam = EvaluateRuleParam(List(rule))
+    )
+
+    dqJob.execute(dqContext)
+
+    spark.sql(s"select * from ${rule.getOutDfName()}")
+  }
+
+  private def createPersonTable(): Unit = {
+    val personCsvPath = getClass.getResource("/hive/person_table.csv").getFile
+
+    spark.sql(
+      s"CREATE TABLE IF NOT EXISTS ${PERSON_TABLE} " +
+        "( " +
+        "  name String," +
+        "  age int " +
+        ") " +
+        "ROW FORMAT DELIMITED FIELDS TERMINATED BY ',' LINES TERMINATED BY '\n' " +
+        "STORED AS TEXTFILE"
+    )
+
+    spark.sql(s"LOAD DATA LOCAL INPATH '$personCsvPath' OVERWRITE INTO TABLE ${PERSON_TABLE}")
+  }
+
+  private def createEmptyPersonTable(): Unit = {
+    spark.sql(
+      s"CREATE TABLE IF NOT EXISTS ${EMPTY_PERSON_TABLE} " +
+        "( " +
+        "  name String," +
+        "  age int " +
+        ") " +
+        "ROW FORMAT DELIMITED FIELDS TERMINATED BY ',' LINES TERMINATED BY '\n' " +
+        "STORED AS TEXTFILE"
+    )
+
+    spark.sql(s"select * from ${EMPTY_PERSON_TABLE}").show()
```

Comment:
assert on emptiness would be useful, what do you think?

Reply:
yeah. Fixed it with drop tables.